cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -290,10 +290,10 @@ func (g *gitHubRepository) getVersions() ([]string, error) {
 	// NB. currently Github API does not support result ordering, so it not possible to limit results
 	var allReleases []*github.RepositoryRelease
 	var retryError error
-	_ = wait.PollImmediate(retryableOperationInterval, retryableOperationTimeout, func() (bool, error) {
+	_ = wait.PollUntilContextTimeout(context.TODO(), retryableOperationInterval, retryableOperationTimeout, true, func(ctx context.Context) (bool, error) {
```

Comment:
nit-ish: is context.Background() a wee bit better here?

Reply:
cc @nawazkh we probably want to audit usages of these deprecated apimachinery wait conveniences in CAPZ as well to be consistent w/ CAPI.